airbyte-cdk/python/airbyte_cdk/sources/declarative/types.py

```diff
@@ -6,11 +6,53 @@
 
 from typing import Any, List, Mapping
 
-Record = Mapping[str, Any]
 # A FieldPointer designates a path to a field inside a mapping. For example, retrieving ["k1", "k1.2"] in the object {"k1" :{"k1.2":
 # "hello"}] returns "hello"
 FieldPointer = List[str]
 Config = Mapping[str, Any]
 ConnectionDefinition = Mapping[str, Any]
 StreamSlice = Mapping[str, Any]
 StreamState = Mapping[str, Any]
+
+
+class Record(Mapping[str, Any]):
+    def __init__(self, data: dict, associated_slice: StreamSlice):
+        self._data = data
+        self._associated_slice = associated_slice
+
+    @property
+    def data(self) -> dict:
+        return self._data
+
+    @property
+    def associated_slice(self) -> StreamSlice:
+        return self._associated_slice
+
+    def __repr__(self):
+        return repr(self._data)
+
+    def __setitem__(self, key: str, value: Any):
+        self._data[key] = value
+
+    def __getitem__(self, key: str):
+        return self._data[key]
+
+    def __len__(self):
+        return len(self._data)
+
+    def __iter__(self):
+        return iter(self._data)
+
+    def __contains__(self, item: str):
+        return item in self._data
+
+    def __eq__(self, other):
+        if isinstance(other, dict):
```

Comment:
I was wondering if the comparison with a dict is relevant here. I did implement this for backward compatibility but yeah...